.github/workflows/ci.yaml

```
@@ -304,3 +342,69 @@ jobs:
       run: make install-complete-requirements
     - name: Install All Requirements ${{ matrix.python-version }}
       run: make install-all-requirements
+
+  coverage:
+    name: Full coverage
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.9"]  # coverage running on 3.9 only right now
+    needs: [tests, integration-tests, migrations-tests]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: pip
+      - name: Install dependencies
+        # To save time, install only pytest and coverage-related packages, without mlrun and its dependencies.
+        run: make install-dev-requirements
+      # File name is not effected by the artifact name (=identifier only)
+      - name: Download server unit tests coverage report
+        uses: actions/download-artifact@v4
+        with:
+          name: coverage_unit_tests_report_${{ matrix.python-version }}_${{ github.run_id }}_server.coverage
+          # Where to download. file downloaded by name.
+          path: tests/coverage_reports/
+      - name: Download client unit tests coverage report
+        uses: actions/download-artifact@v4
+        with:
+          name: coverage_unit_tests_report_${{ matrix.python-version }}_${{ github.run_id }}_client.coverage
+          # Where to download. file downloaded by name.
+          path: tests/coverage_reports/
+      - name: Download integration coverage report
+        uses: actions/download-artifact@v4
+        with:
+          name: coverage_integration_tests_report_${{ matrix.python-version }}_${{ github.run_id }}.coverage
+          # Where to download. file downloaded by name.
+          path: tests/coverage_reports/
+
+      - name: Download migration coverage report
+        uses: actions/download-artifact@v4
+        with:
+          name: coverage_migration_tests_report-${{ github.run_id }}.coverage
+          # Where to download. file downloaded by name.
+          path: tests/coverage_reports/
+
+      - name: Combine coverage
+        run: make coverage-combine UNIT_TEST_COVERAGE_PATHS="tests/coverage_reports/unit_tests_client.coverage tests/coverage_reports/unit_tests_server.coverage"
+      # save it as an artifact:
```

I think the indentation should be where the `name` starts instead of the hyphen
```suggestion
        # Save it as an artifact:
```